pkg/storages/storage/folder.go

```diff
@@ -99,11 +84,14 @@ func ListFolderRecursivelyWithFilter(folder Folder, folderSelector func(path str
 	return relativePathObjects, nil
 }
 
-func addPrefixToNames(objects []Object, folderPrefix string) []Object {
+func makePathsRelative(objects []Object, folderPrefix string) []Object {
```

Comment:
This function is called from different `ListFolderRecursively*` methods. And this functions replaces names of the received objects to the relative paths, which may be longer than just names.

This doesn't work with multistorage folders, because when we create a new instance of `storage.LocalObject` struct, we loose the information about the storage where this object is stored.

So here I introduced a special `RelativePathObject` struct, that implements both `storage.Folder` and `multistorage.MultiObject` interfaces and just wraps any other `storage.Folder` calculating the relative path on flight. 

I also don't think that storing paths in `storage.Object.GetName()` is a good idea because it's always not clear from the code what is stored in some specific object - just the name or the path. Maybe we need to rework it somehow. Now I introduced at least a specific implementation for that.